src/engine/app.rs

```diff
@@ -3,25 +3,55 @@
 use num_cpus;
 use std::time::{Duration, Instant};
 
-use asset_manager::AssetManager;
 use ecs::{Component, Planner, Priority, System, World};
 use ecs::components::{LocalTransform, Transform, Child, Init, Renderable};
 use ecs::resources::Time;
 use ecs::systems::TransformSystem;
 use engine::state::{State, StateMachine};
 use engine::timing::Stopwatch;
 use gfx_device;
-use gfx_device::{DisplayConfig, GfxDevice, gfx_types};
+use gfx_device::{DisplayConfig, GfxDevice};
+use gfx_device::gfx_types::Factory;
 use renderer::{AmbientLight, DirectionalLight, Pipeline, PointLight, target};
 
+/// A context, which stores structs
+/// that are required for asset instantiation,
+/// like the gfx factory.
+pub struct Context {
```

Comment:
Maybe expose `Factory` directly? It has a more descriptive name and It is `pub` anyway. I am not sure if something like a `Factory` would be requried for loading audio files. For example `lewton` (ogg loading library) doesn't need it, you just create an [`OggStreamReader`](https://github.com/RustAudio/lewton/blob/master/examples/perf.rs#L28) from a file.